File: stripe/src/test/java/com/stripe/android/PayWithGoogleUtilsTest.java
package com.stripe.android;

import org.junit.After;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.RobolectricTestRunner;

import java.util.Currency;
import java.util.Locale;

import static com.stripe.android.PayWithGoogleUtils.getPriceString;
import static org.junit.Assert.assertEquals;

/**
 * Test class for {@link PayWithGoogleUtils}.
 */
@RunWith(RobolectricTestRunner.class)
public class PayWithGoogleUtilsTest {

    @After
    public void tearDown() {
        Locale.setDefault(Locale.US);
    }

    @Test
    public void getPriceString_whenCurrencyWithDecimals_returnsExpectedValue() {
        String priceString = getPriceString(100L, Currency.getInstance("USD"));
        assertEquals("1.00", priceString);

        String littlePrice = getPriceString(8L, Currency.getInstance("EUR"));
        assertEquals("0.08", littlePrice);

        String bigPrice = getPriceString(20000000L, Currency.getInstance("GBP"));
        assertEquals("200000.00", bigPrice);
    }

    @Test
    public void getPriceString_whenLocaleWithCommas_returnsExpectedValue() {
        Locale.setDefault(Locale.FRENCH);
        String priceString = getPriceString(100L, Currency.getInstance("USD"));
        assertEquals("1.00", priceString);

        String littlePrice = getPriceString(8L, Currency.getInstance("EUR"));
        assertEquals("0.08", littlePrice);

        String bigPrice = getPriceString(20000000L, Currency.getInstance("GBP"));
        assertEquals("200000.00", bigPrice);
    }

    @Test
    public void getPriceString_whenCurrencyWithoutDecimals_returnsExpectedValue() {
        String priceString = getPriceString(250L, Currency.getInstance("JPY"));
        assertEquals("250", priceString);

        String bigPrice = getPriceString(250000L, Currency.getInstance("KRW"));
        assertEquals("250000", bigPrice);

        String littlePrice = getPriceString(7L, Currency.getInstance("CLP"));
        assertEquals("7", littlePrice);
    }
}

File: stripe/src/test/java/com/stripe/android/PayWithGoogleUtilsTest.kt
package com.stripe.android

import com.stripe.android.PayWithGoogleUtils.getPriceString
import java.util.Currency
import java.util.Locale
import kotlin.test.AfterTest
import kotlin.test.Test
import kotlin.test.assertEquals

/**
 * Test class for [PayWithGoogleUtils].
 */
class PayWithGoogleUtilsTest {

    @AfterTest
    fun tearDown() {
        Locale.setDefault(Locale.US)
    }

    @Test
    fun getPriceString_whenCurrencyWithDecimals_returnsExpectedValue() {
        val priceString = getPriceString(100, Currency.getInstance("USD"))
        assertEquals("1.00", priceString)

        val littlePrice = getPriceString(8, Currency.getInstance("EUR"))
        assertEquals("0.08", littlePrice)

        val bigPrice = getPriceString(20000000, Currency.getInstance("GBP"))
        assertEquals("200000.00", bigPrice)
    }

    @Test
    fun getPriceString_whenLocaleWithCommas_returnsExpectedValue() {
        Locale.setDefault(Locale.FRENCH)
        val priceString = getPriceString(100, Currency.getInstance("USD"))
        assertEquals("1.00", priceString)

        val littlePrice = getPriceString(8, Currency.getInstance("EUR"))
        assertEquals("0.08", littlePrice)

        val bigPrice = getPriceString(20000000, Currency.getInstance("GBP"))
        assertEquals("200000.00", bigPrice)
    }

    @Test
    fun getPriceString_whenCurrencyWithoutDecimals_returnsExpectedValue() {
        val priceString = getPriceString(250, Currency.getInstance("JPY"))
        assertEquals("250", priceString)

        val bigPrice = getPriceString(250000, Currency.getInstance("KRW"))
        assertEquals("250000", bigPrice)

        val littlePrice = getPriceString(7, Currency.getInstance("CLP"))
        assertEquals("7", littlePrice)
    }
}
